src/sentry/tasks/embeddings_grouping/utils.py

```diff
@@ -106,13 +111,28 @@ def initialize_backfill(project_id, last_processed_index):
 
     redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
 
-    if last_processed_index is None:
-        last_processed_index = int(redis_client.get(make_backfill_redis_key(project_id)) or 0)
-    return project, redis_client, last_processed_index
+    if last_processed_group_index is None:
+        last_processed_group_index_ret = int(
+            redis_client.get(make_backfill_grouping_index_redis_key(project_id)) or 0
+        )
+    else:
+        last_processed_group_index_ret = last_processed_group_index
+
+    if last_processed_project_index is None:
+        if cohort and isinstance(cohort, str):
+            last_processed_project_index_ret = int(
+                redis_client.get(make_backfill_project_index_redis_key(cohort, project_id)) or 0
```

Comment:
what value would this be getting? I only see us setting it using `last_processed_project_index` instead of `project_id`

Reply:
ah great catch! 🙌🏼 thank you